ui/app/pages/mobile-sync/mobile-sync.component.js

```diff
@@ -0,0 +1,436 @@
+import React, { Component } from 'react'
+
+const PropTypes = require('prop-types')
+const classnames = require('classnames')
+const PubNub = require('pubnub')
+const qrCode = require('qrcode-generator')
+
+const { DEFAULT_ROUTE } = require('../../helpers/constants/routes')
+
+import Button from '../../components/ui/button'
+import LoadingScreen from '../../components/ui/loading-screen'
+
+const PASSWORD_PROMPT_SCREEN = 'PASSWORD_PROMPT_SCREEN'
+const REVEAL_SEED_SCREEN = 'REVEAL_SEED_SCREEN'
+const KEYS_GENERATION_TIME = 30000
+
+export default class MobileSyncPage extends Component {
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    history: PropTypes.object,
```

Comment:
It looks like these props should all be `isRequired`.

Reply:
Whoops, I'll update this in a new PR, got a bit eager